source/frontend/src/features/leases/shared/propertyPicker/LeasePropertySelector.tsx

```diff
@@ -0,0 +1,167 @@
+import SelectedPropertyHeaderRow from './selectedPropertyList/SelectedPropertyHeaderRow';
+import SelectedPropertyRow from './selectedPropertyList/SelectedPropertyRow';
+
+interface LeasePropertySelectorProp {
+  formikProps: FormikProps<LeaseFormModel>;
+}
+
+export const LeasePropertySelector: React.FunctionComponent<
+  React.PropsWithChildren<LeasePropertySelectorProp>
+> = ({ formikProps }) => {
+  const { values } = formikProps;
+
+  const { getProperties } = useProperties();
+
+  const { setModalContent, setDisplayModal } = useContext(ModalContext);
+  const [propertiesToConfirm, setPropertiesToConfirm] = useState<FormLeaseProperty[]>([]);
+
+  const arrayHelpersRef = useRef<FieldArrayRenderProps | null>(null);
+
+  const addProperties = useCallback(
+    (properties: FormLeaseProperty[]) => {
+      if (arrayHelpersRef.current !== null && properties.length > 0) {
+        properties.forEach(property => {
+          arrayHelpersRef.current!.push(property);
+        });
+      }
+    },
+    [arrayHelpersRef],
+  );
+
+  const searchProperty = async (newProperty: IMapProperty): Promise<IProperty[] | undefined> => {
+    const params: IPropertyFilter = {
+      pinOrPid: (newProperty.pid || newProperty.pin || '')?.toString(),
+      searchBy: 'pinOrPid',
+      address: '',
+    };
+
+    const result = await getProperties.execute(params);
+    return result?.items;
+  };
+
+  const confirmAdd = useCallback(() => {
+    setDisplayModal(false);
+    addProperties(propertiesToConfirm);
+    setPropertiesToConfirm([]);
+  }, [setDisplayModal, addProperties, propertiesToConfirm]);
+
+  const cancelAdd = useCallback(() => {
+    setDisplayModal(false);
+    setPropertiesToConfirm([]);
+  }, [setDisplayModal]);
+
+  const customModalProps: ModalProps = useMemo(() => {
+    return {
+      title: 'Not inventory property',
+      message:
+        'You have selected a property not previously in the inventory. Do you want to add this property to the lease?',
+      display: false,
+      closeButton: false,
+      okButtonText: 'Add',
+      cancelButtonText: 'Cancel',
+      handleOk: confirmAdd,
+      handleCancel: cancelAdd,
+    };
+  }, [confirmAdd, cancelAdd]);
+
+  useDeepCompareEffect(() => {
+    setModalContent(customModalProps);
+  }, [customModalProps]);
+
+  return (
+    <Section header="Properties to include in this file:">
+      <div className="py-2">
+        Select one or more properties that you want to include in this lease/license file. You can
+        choose a location from the map, or search by other criteria.
+      </div>
+
+      <FieldArray
+        name="properties"
+        render={arrayHelpers => {
+          arrayHelpersRef.current = arrayHelpers;
+          return (
+            <>
+              <Row className="py-3 no-gutters">
+                <Col>
+                  <MapSelectorContainer
+                    addSelectedProperties={async (newProperties: IMapProperty[]) => {
```

Comment:
This is a pretty big function to have embedded in jsx, would lean towards having this in a function outside of the jsx